packages/flame/lib/src/sprite.dart

```diff
@@ -81,6 +81,10 @@ class Sprite {
     );
   }
 
+  // Used to avoid the creation of new Vector2 objects in render.
+  late final _tmpRenderPosition = Vector2.zero();
+  late final _tmpRenderSize = Vector2.zero();
```

Comment:
Probably no need to make them `late`.

Reply:
No harm in doing so right? If some user has some big collection of sprites that aren't being rendered yet they have a little bit less overhead.

Reply:
Well, the overhead is that there needs to be a runtime check whether the `late` variable has actually been initialized or not. And I'm not sure whether such a check is debug-mode only, or if it is also done in production...

Reply:
I don't think such a check is even done in debug-mode? Then we'd have better stack traces for `LateInitializationError` right?

Reply:
That must be very minimal? But I don't know how it is done.
I'd say it is worth the trade-off of not carrying around two `Vector2` before they are used.
I can do some benchmarking tomorrow.

Reply:
Another possibility is to declare them `static`. Then you have only one object instance no matter how many sprites

Reply:
True, single thread ftw 😄